src/dbms/dbms_handler.hpp

```diff
@@ -131,8 +133,52 @@ class DbmsHandler {
    */
   NewResultT New(const std::string &name) {
     std::lock_guard<LockT> wr(lock_);
-    return New_(name, name);
+    const auto uuid = utils::UUID{};
+    return New_(name, uuid);
   }
+  /**
+   * @brief Create a new Database using the passed configuration
+   *
+   * @param config configuration to be used
+   * @return NewResultT context on success, error on failure
+   */
+  NewResultT New(const storage::SalientConfig &config) {
+    std::lock_guard<LockT> wr(lock_);
+    auto config_copy = default_config_;
+    config_copy.salient = config;  // name, uuid, mode, etc
+    UpdatePaths(config_copy, config_copy.durability.storage_directory / "database" / std::string{config.uuid});
+    auto new_db = New_(config_copy);
+
+    if (repl_state_.IsReplica()) {
+      // If database exists, update the UUID
+      if (new_db.HasError() && new_db.GetError() == NewError::EXISTS) {
+        spdlog::debug("Trying to create db '{}' on replica which already exists.", config.name);
+        auto db = Get_(config.name);
+        if (db->uuid() != config.uuid) {
+          spdlog::debug("Different UUIDs");
+
+          // TODO: Fix this hack
+          if (config.name == kDefaultDB) {
+            spdlog::debug("Update default db's UUID");
+            // Default db cannot be deleted and remade, have to just update the UUID
+            db->storage()->config_.salient.uuid = config.uuid;
+            UpdateDurability(db->storage()->config_, ".");
+            return db;
+          }
```

Comment:
What is writes already exist in this other default database. I would rather wipe the whole default database, giving it new storage.

Reply:
Or some kind of check that the database is unused. ie. Check timestamp is 0.

Reply:
We now make sure the database is clean before updating UUID.
It will fail otherwise.